coucharchive

```diff
@@ -264,15 +265,15 @@ class ReplicationControl(object):
                                   self.running_replications.value))
 
     def _drop_old_events(self):
-        five_min_ago = time.time() - 5 * 60
+        ten_min_ago = time.time() - 10 * 60
         self._last_successes[:] = [e for e in self._last_successes
-                                   if e[0] > five_min_ago]
+                                   if e[0] > ten_min_ago]
         self._last_errors[:] = [e for e in self._last_errors
-                                if e[0] > five_min_ago]
+                                if e[0] > ten_min_ago]
 
     def recent_errors(self):
-        self._drop_old_events()
-        return len(self._last_errors)
+        five_min_ago = time.time() - 5 * 60
```

Comment:
It's strange to keep 5 min at some places, and 10 at others. Would you be OK for 10 minutes everywhere, and just keep `wait = 5 * len(self._last_errors)` in `def ideal_sleep_value()`?

Reply:
Ok!
You don't fear that the effect of `ideal_sleep_value()` would become to much?

Reply:
PS: new diff with 10 minutes for all, and fixup squashed:
```diff
--- a/coucharchive
+++ b/coucharchive
@@ -12,6 +12,7 @@ import getpass
 from io import BytesIO
 import json
 import logging
+import math
 import multiprocessing
 import os
 import queue
@@ -264,11 +265,11 @@ class ReplicationControl(object):
                                   self.running_replications.value))
 
     def _drop_old_events(self):
-        five_min_ago = time.time() - 5 * 60
+        ten_min_ago = time.time() - 10 * 60
         self._last_successes[:] = [e for e in self._last_successes
-                                   if e[0] > five_min_ago]
+                                   if e[0] > ten_min_ago]
         self._last_errors[:] = [e for e in self._last_errors
-                                if e[0] > five_min_ago]
+                                if e[0] > ten_min_ago]
 
     def recent_errors(self):
         self._drop_old_events()
@@ -320,7 +321,7 @@ class ReplicationControl(object):
         #           +      |   |
         #    +    + +      +   +
         #  ----------------------> t
-        #        5 minutes
+        #        10 minutes
         now = time.time()
         weights = []
         weighted_values = []
@@ -331,8 +332,8 @@ class ReplicationControl(object):
         for i in last_errors_lowered + target:
             age = now - i[0]
             # Weight for an old event is 0, weight for a recent event is 1,
-            # between the two it is a linear curve.
-            weight = max(0, 1 - age / (5 * 60))
+            # between the two it is an exponential curve.
+            weight = math.exp(- age / 120)
             weights.append(weight)
             weighted_values.append(weight * i[1])
         ideal_number = sum(weighted_values) / sum(weights)

Reply:
> You don't fear that the effect of `ideal_sleep_value()` would become to much?

It could indeed, but it would *double* at max, and is still limited to 1 minute maximum. So I'm not afraid!

> an exponential curve.

+1